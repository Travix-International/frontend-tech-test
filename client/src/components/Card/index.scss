.card{
    box-sizing: border-box;
    padding: 1.5rem 0;
    border-radius: 5px;
    box-shadow: 0 3px 15px 1px rgba(0, 0, 0, 0.15);
    margin-bottom: 1.2rem;
}