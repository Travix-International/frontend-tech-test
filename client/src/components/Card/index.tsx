import React from 'react';
import styles from './index.scss';

const Card: React.FC<{}> = ({children}) => (
    <div className={styles.card}>
        {children}
    </div>
);

export default Card;