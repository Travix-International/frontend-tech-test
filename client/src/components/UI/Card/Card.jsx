import React from 'react';
import PropTypes from 'prop-types';

import styles from './Card.module.scss';

const Card = ({children}) => (
  <div className={styles.Card}>
    {children}
  </div>
);

Card.propTypes = {
  children: PropTypes.node.isRequired
}

export default Card;