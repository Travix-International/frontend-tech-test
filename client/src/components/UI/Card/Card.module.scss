.Card {
  background: #fff;
  border-radius: 10px;
  margin-top: 20px;
  padding: 20px;
}