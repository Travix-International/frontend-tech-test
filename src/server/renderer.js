import React from 'react';
import ReactServer from 'react-dom/server';
import { StaticRouter as Router, matchPath } from 'react-router';
import { Provider } from 'react-redux';
import { Helmet } from 'react-helmet';
import { /* get as _get, */ isFunction as _isFunction } from 'lodash';
import renderFullPage from './layout';
import configureStore from '../configureStore';
import { entryRouteComponentMap, getEntryAndRoute, entryReducerMap } from './entryAndRoute';

function applyInitStore(req) {
  const entryRouteInfo = getEntryAndRoute(req.path);
  const currentEntry = entryRouteInfo.entry;
  const rootReducer = entryReducerMap[currentEntry];
  const store = configureStore({}, rootReducer);
  req.reduxStore = store;
  req.entryRouteInfo = entryRouteInfo;
  // next();
}

function applyRouteCheckResult(req) {
  const { path, reduxStore: store, query, entryRouteInfo } = req;
  const promises = [];
  const currentEntry = entryRouteInfo.entry;
  const route = entryRouteInfo.route;
  const match = matchPath(path, route);
  const { loadData } = route;

  // set currentEntry to req
  req.currentEntry = currentEntry;

  // it's possible no loadData set
  if (_isFunction(loadData)) {
    let action = loadData(match, query);
    if (action) {
      if (_isFunction(action) || action.type) {
        action = store.dispatch(action);
        if (action.then) {
          promises.push(action);
        }
      }
    }
  }

  return Promise.all(promises);
}

function responsePage(req, res, clientStats) {
  const { url, reduxStore: store, currentEntry } = req;
  const routerContext = {};
  const reduxStateString = JSON.stringify(store.getState()).replace(/</g, '\\u003c');

  const RouteComponent = entryRouteComponentMap[currentEntry];

  const content = (
    <Provider store={store}>
      <Router context={routerContext} location={url}>
        <RouteComponent />
      </Router>
    </Provider>
  );

  // render to sting to get helmet setting
  const contentString = ReactServer.renderToString(content);
  const head = Helmet.renderStatic();
  const htmlString = renderFullPage(contentString, reduxStateString, head, currentEntry, clientStats);

  res.send(htmlString);
}

export default function renderer({ clientStats }) {
  return (req, res) => {
    applyInitStore(req);
    return applyRouteCheckResult(req).then(() => {
      return responsePage(req, res, clientStats);
    });
  };
}
