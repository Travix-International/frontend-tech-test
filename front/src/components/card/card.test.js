import React from 'react'
import renderer from 'react-test-renderer'

import Card from '.'

describe('CardComponent', () => {
  it('Snapshot', async () => {
    const tree = renderer
      .create(<Card><p>test</p></Card>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
