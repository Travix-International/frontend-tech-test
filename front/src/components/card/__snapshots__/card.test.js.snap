// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardComponent Snapshot 1`] = `
.c0 {
  position: relative;
  width: 100%;
  height: calc(100vh - 7rem);
  background: #fff;
  border-radius: .4rem;
  min-width: 27.5rem;
  box-sizing: border-box;
  padding-bottom: 2.5rem;
}

<div
  className="c0"
>
  <p>
    test
  </p>
</div>
`;
